app/controllers/application_controller.rb

```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  include Pundit
+
+  protect_from_forgery
+  before_action :authenticate_user!
+  after_action :verify_authorized
```

Comment:
```suggestion
  after_action :verify_authorized
```
Are you sure about `after_action`? BTW I highly recommend avoiding keeping too much logic using callbacks, it could be really hard to understand what a particular endpoint does